pwndbg/arguments.py

```diff
@@ -158,14 +159,36 @@ def arguments(abi: pwndbg.lib.abi.ABI | None = None):
         yield argname(i, abi), argument(i, abi)
 
 
+# When an argument is named one of these in Linux syscalls/glibc, it refers to a file descriptor
+# Search for strings containing "fd" in https://chromium.googlesource.com/chromiumos/docs/+/master/constants/syscalls.md
+FILE_DESCRIPTOR_ARG_NAMES = {
+    "fd",
+    "in_fd",
+    "out_fd",
+    "fdin",
+    "fdout",
+    "oldfd",
+    "fildes",
+    "newfd",
+    "epfd",
+    "dfd",
+    "dirfd",
+    "mountdirfd",
+}
+
+
 def format_args(instruction: PwndbgInstruction) -> List[str]:
     result = []
     for arg, value in get(instruction):
         code = arg.type != "char"
-        pretty = pwndbg.chain.format(value, code=code)
+        pretty = (
+            pwndbg.chain.format(value, code=code)
+            if not arg.flags
+            else format_flags_argument(arg.flags, value)
+        )
 
         # Enhance args display
-        if arg.name == "fd" and isinstance(value, int):
+        if arg.name in FILE_DESCRIPTOR_ARG_NAMES and isinstance(value, int):
```

Comment:
On a long run such enhancing should probably be done on a per-function basis instead of here ( ? ).

Maybe we shall create an issue for that as 'code cleanup' idk...

We could attach such 'enhancers' or display helpers to the function objects or whatever (?). Loud thinking here.